x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformIndex.java

```diff
@@ -134,10 +148,28 @@ private static Settings createSettings() {
      * }
      * @param mappings A Map of the form {"fieldName": "fieldType"}
      */
-    private static Map<String, Object> createMappingsFromStringMap(Map<String, String> mappings) {
+    static Map<String, Object> createMappingsFromStringMap(Map<String, String> mappings) {
+        List<Map.Entry<String, String>> sortedMappingsEntries = new ArrayList<>(mappings.entrySet());
+        // We sort the entry list to make sure that for each (parent, parent.child) pair, parent entry will be processed before child entry.
+        sortedMappingsEntries.sort(comparingByKey());
         Map<String, Object> fieldMappings = new HashMap<>();
-        mappings.forEach((k, v) -> fieldMappings.put(k, Map.of("type", v)));
-
+        for (Map.Entry<String, String> entry : sortedMappingsEntries) {
+            String[] parts = entry.getKey().split("\\.");
```

Comment:
nit: I am not a big fan of using regex if there is no reason. Arguably its java's fault. A potential alternative: the tokenize methods in `org.elasticsearch.common.Strings`

Reply:
> Arguably its java's fault.

Unfortunately I couldn't find a non-regex version of `split()` method.

> A potential alternative: the tokenize methods in org.elasticsearch.common.Strings

Thanks! I've used `Strings.tokenizeToStringArray` method.